crates/pathfinder/src/rpc/types.rs

```diff
@@ -106,15 +106,10 @@ pub mod request {
         #[serde(default)]
         pub keys: Vec<EventKey>,
 
-        #[serde(flatten)]
-        pub pagination: Option<ResultPageRequest>,
-    }
-
-    #[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
-    #[serde(deny_unknown_fields)]
-    pub struct ResultPageRequest {
-        pub page_size: usize,
-        pub page_number: usize,
+        // These are inlined here because serde flatten and deny_unknown_fields
+        // don't work together.
+        pub page_size: Option<usize>,
+        pub page_number: Option<usize>,
```

Comment:
This does mean we would accept incorrect permutations though correct? For example, a `page_size` without a `page_number` specified. Or vice versa.

I guess we can just say that either missing effectively means "both" are missing.

Looking at the (incorrect `allof`) spec -- should we really be flattening this though? Is the query not supposed to look something like:
```rust
struct Query {
    filter: struct {
        from_block, to_block, address, address,
    }, 
    pagination: Option<struct {
        page_size, page_number
    }>
}
```

Reply:
I don't think the query is supposed to look like this according to the current spec:

```
            "name": "starknet_getEvents",
            "summary": "Returns all events matching the given filter",
            "description": "Returns all event objects matching the conditions in the provided filter",
            "params": [
                {
                    "name": "filter",
                    "summary": "The conditions used to filter the returned events",
                    "required": true,
                    "schema": {
                        "allOf": [
                            {
                                "$ref": "#/components/schemas/EVENT_FILTER"
                            },
                            {
                                "$ref": "#/components/schemas/RESULT_PAGE_REQUEST"
                            }
                        ]
                    }
                }
            ],
```

Here `filter` is just the name of the parameter, and both EVENT_FILTER and RESULT_PAGE_REQUESTS is just an object with a few properties (none of those required).

Maybe we could just ask for a modification here so that `filter` and `pagination` are _two_ params instead of flattening those into a single one?

Reply:
Yes but you're looking at `allof` as being used somewhat correctly -- I think it is being used to indicate the the schema should contain **_all of_** the following fields.. i.e. an `EVENT_FILTER` and a `RESULT_PAGE_REQUEST`. The alternative is that those fields must always exist -- so no `Option` at all. Which I would also be okay with.

Reply:
Yeah, it'd be great to have a spec which makes this clear ;)